android/sdl_android/src/main/java/com/smartdevicelink/util/AndroidTools.java

```diff
@@ -191,4 +196,31 @@ public static Bitmap downloadImage(String urlStr) throws IOException {
 		bis.close();
 		return result;
 	}
+
+	/**
+	 * This is a wrapper around the startForegroundService method. In the
+	 * event that the user is on Android 28+ it will check the FOREGROUND_SERVICE permissions
+	 * before trying to call startForegroundService.
+	 * @param context a context instance
+	 * @param intent the foreground service intent
+	 * @return a ComponentName, an identifier for the started service, will return null is service
+	 * was unable to start
+	 */
+	@RequiresApi(api = Build.VERSION_CODES.O)
+	public static ComponentName safeStartForegroundService(Context context, Intent intent) {
+		if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.P) {
+			boolean inDebugMode = (0 != (context.getApplicationInfo().flags & ApplicationInfo.FLAG_DEBUGGABLE));
+			boolean permissionGranted = PackageManager.PERMISSION_GRANTED == context.checkPermission(Manifest.permission.FOREGROUND_SERVICE, android.os.Process.myPid(), android.os.Process.myUid());
+			if (inDebugMode || permissionGranted) {
+				return context.startForegroundService(intent);
+			} else {
+				DebugTool.logError("App missing FOREGROUND_SERVICE Permissions");
+			}
+		} else if (Build.VERSION.SDK_INT == Build.VERSION_CODES.O) {
+			return context.startForegroundService(intent);
+		} else if (Build.VERSION.SDK_INT < Build.VERSION_CODES.O) {
+			DebugTool.logError("startForegroundService() call requires Android Oreo or newer");
+		}
```

Comment:
I have just learned that there are two versions of Android O (`VERSION_CODES.O`= `26` & `VERSION_CODES.O_MR1` = `27`) and they both don't require the foreground permission. So I updated the code accordingly 

```suggestion
		} else if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
			return context.startForegroundService(intent);
		}
```